### ISSUE TITLE
Bluetooth still ocasionally not available after reboot, with R2 Release Candidate

### ISSUE THREAD

**Issue by JunOllyLi** [ev3dev/ev3dev#1314]
With ev3dev-stretch R2 Release Candidate, snapshot-ev3dev-stretch-ev3-generic-2019-09-08.img.xz
I have to say the bluetooth is a lot more stable than before. But I still encountered the problem once.
After starting up, BT is not available, and it went away after a reboot.

**Comment by philltran** [comment]
I am experiencing the same on R2,. Is rebooting the only work around?

**Comment by dlech** [comment]
It depends on where the failure is. One time, I was able to run `sudo systemctl restart bluetooth` to get it working with out reboot, but another time this didn't work.

We could also try experimenting with the sleep time to see if making it longer makes a difference.

    sudo nano /lib/systemd/system/ev3-bluetooth.service

And change the line:

    ExecStartPre=/bin/sleep 10

**Comment by philltran** [comment]
Thanks for the tip. Let me know if there is anything I can do to help with the project.

**Comment by dlech** [comment]
Try 20 or 30 seconds and report back the results :smile:

**Comment by philltran** [comment]
Of course now it's working after I re-imaged the SD card. 

I changed the sleep time and everything seems to still be working. 

I will try it again if the bluetooth goes out again. 

Thanks for the help.

**Comment by keenanjohnson** [comment]
I'm having this issue on my ev3 as well.

**Comment by keenanjohnson** [comment]
Updating the sleep time to 30 did not resolve the issue for me.

**Comment by keenanjohnson** [comment]
Interestingly reimaging the SD card has also resolved this for me. Seems like currently the image process must be done twice?

**Comment by dlech** [comment]
Replying to @JanderII https://github.com/ev3dev/ev3dev/issues/925#issuecomment-550994759 and https://github.com/ev3dev/ev3dev/issues/925#issuecomment-551030963

I've seen some other people mention that installing/fiddling with `rfkill` seems to make a difference. Theoretically, `connmand` is supposed to take care of this and installing `rfkill` should not be necessary.

Could everyone who is still having problems share what kind of SD card they are using? I am starting to wonder if cheap (and therefore slow) microSD cards are affecting the timing compared to the SanDisk Extreme cards that I am using.

**Comment by JanderII** [comment]
I'm using a SanDisk Ultra Class 10 A1 32GB.

It is certainly bizarre that after lots of failures that it's now consistently working. That does seem to imply that something has changed on the SD card now.

**Comment by keenanjohnson** [comment]
Yeah I'm also using a fairly high quality SD card. A SanDisk Ultra Class 10.

https://www.amazon.com/SanDisk-microSDHC-Standard-Packaging-SDSQUNC-032G-GN6MA/dp/B010Q57T02

**Comment by neilmcguire** [comment]
I was having Bluetooth Not Available issues consistently, then made sure I could connect to the BT in Lego OS so I knew it wasn't a hardware issue.

I am using a Kingston 32 GB Class 10. We followed the `rfkill` suggestions over on ev3dev/ev3dev#925 and it made the Bluetooth available. Because the reflash has fixed it for some people I am trying that right now, and will report back. And if it doesn't work to reflash, at least I have the `rfkill` fiddling I can do.

UPDATE: Reflashing the SD card also solved the problem.

Here's what it looked like:
```
robot@ev3dev:~$ ev3dev-sysinfo
Image file:         ev3dev-stretch-ev3-generic-2019-10-23
Kernel version:     4.14.117-ev3dev-2.3.4-ev3
Brickman:           0.10.2
BogoMIPS:           148.88
Bluetooth:
Board:              board0
BOARD_INFO_HW_REV=7
BOARD_INFO_MODEL=LEGO MINDSTORMS EV3
BOARD_INFO_ROM_REV=6
BOARD_INFO_SERIAL_NUM=0016534C6247
BOARD_INFO_TYPE=main
robot@ev3dev:~$ sudo apt-get update && sudo apt-get install rfkill
robot@ev3dev:~$ /usr/bin/hciattach -t 30 /dev/ttyS2 any 2000000 flow nosleep
robot@ev3dev:~$ sudo rfkill list all
0: phy0: Wireless LAN
        Soft blocked: no
        Hard blocked: no
2: hci0: Bluetooth
        Soft blocked: yes
        Hard blocked: no
robot@ev3dev:~$ sudo rfkill unblock 2
robot@ev3dev:~$ sudo rfkill list all
0: phy0: Wireless LAN
        Soft blocked: no
        Hard blocked: no
2: hci0: Bluetooth
        Soft blocked: no
        Hard blocked: no
robot@ev3dev:~$ sudo reboot
```

**Comment by salamanders** [comment]
+1 to this issue.
micropython ISO from the Lego site, ran `sudo apt-get dist-upgrade`
Booted with nothing plugged in: no bluetooth.
Booted with just the USB/WiFi dongle in: still no bluetooth.

```
Linux ev3dev 4.14.117-ev3dev-2.3.4-ev3 #1 PREEMPT Thu May 9 15:13:01 CDT 2019 armv5tejl
robot@ev3dev:~$ ev3dev-sysinfo
Image file:         ev3-micropython-v1.0.0-sd-card-image
Kernel version:     4.14.117-ev3dev-2.3.4-ev3
Brickman:           0.10.2
BogoMIPS:           148.88
Bluetooth:          
Board:              board0
BOARD_INFO_HW_REV=7
BOARD_INFO_MODEL=LEGO MINDSTORMS EV3
BOARD_INFO_ROM_REV=6
BOARD_INFO_SERIAL_NUM=00165353FA38
BOARD_INFO_TYPE=main

**Comment by dlech** [comment]
@salamanders What do the logs say? `sudo systemctl --no-pager status ev3-bluetooth` ?

**Comment by CubixSystem** [comment]
@dlech I think my logs may help

```
robot@ev3dev:~$ ev3dev-sysinfo
Image file:         ev3dev-stretch-ev3-generic-2019-03-03
Kernel version:     4.14.117-ev3dev-2.3.4-ev3
Brickman:           0.10.2
BogoMIPS:           226.09
Bluetooth:
Board:              board0
BOARD_INFO_HW_REV=7
BOARD_INFO_MODEL=LEGO MINDSTORMS EV3
BOARD_INFO_ROM_REV=6
BOARD_INFO_SERIAL_NUM=0016534CDB22
BOARD_INFO_TYPE=main
robot@ev3dev:~$ sudo systemctl --no-pager status ev3-bluetooth
[sudo] password for robot:
● ev3-bluetooth.service - LEGO MINDSTORMS EV3 built-in Bluetooth
   Loaded: loaded (/lib/systemd/system/ev3-bluetooth.service; static; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sun 2019-11-24 19:58:20 UTC; 2min 29s ago
  Process: 438 ExecStart=/usr/bin/hciattach -t 30 /dev/ttyS2 texas 2000000 flow nosleep (code=exited, status=1/FAILURE)
  Process: 333 ExecStartPre=/bin/sleep 10 (code=exited, status=0/SUCCESS)

Nov 24 19:57:09 ev3dev systemd[1]: Starting LEGO MINDSTORMS EV3 built-in Bluetooth...
Nov 24 19:57:20 ev3dev hciattach[438]: Found a Texas Instruments' chip!
Nov 24 19:57:20 ev3dev hciattach[438]: Firmware file : /lib/firmware/ti-connectivity/TIInit_6.6.15.bts
Nov 24 19:57:20 ev3dev hciattach[438]: Loaded BTS script version 1
Nov 24 19:57:20 ev3dev hciattach[438]: texas: changing baud rate to 2000000, flow control to 1
Nov 24 19:57:50 ev3dev hciattach[438]: Initialization timed out.
Nov 24 19:58:20 ev3dev systemd[1]: ev3-bluetooth.service: Control process exited, code=exited status=1
Nov 24 19:58:20 ev3dev systemd[1]: Failed to start LEGO MINDSTORMS EV3 built-in Bluetooth.
Nov 24 19:58:20 ev3dev systemd[1]: ev3-bluetooth.service: Unit entered failed state.
Nov 24 19:58:20 ev3dev systemd[1]: ev3-bluetooth.service: Failed with result 'exit-code'.
```

JFYI:
>BogoMIPS:           226.09

I use [overclocked](https://lechnology.com/2018/06/overclocking-lego-mindstorms-ev3-part-2/) ev3

**Comment by dlech** [comment]
> hciattach[438]: Initialization timed out.

This is the old error from #925, so adjusting the delay as describe in https://github.com/ev3dev/ev3dev/issues/1314#issuecomment-537621073 may help in this case.

> ev3dev-stretch-ev3-generic-2019-03-03

Also, this is an old SD card image, so the old error is expected.

**Comment by dlech** [comment]
In this issue, we are expecting a different error, such as:

    hciattach[461]: Can't init device hci0: Operation not possible due to RF-kill (132)Can't initialize device: Operation not possible due to RF-kill

**Comment by CubixSystem** [comment]
@dlech thanks for pointing this out, tomorrow I try `ev3dev-stretch R2` and report results

I'm pretty new here and thought that run [Option 2: Upgrading all installed software](https://www.ev3dev.org/docs/tutorials/upgrading-ev3dev/) should be enough to be up to date

**Comment by dlech** [comment]
> I'm pretty new here and thought that run [Option 2: Upgrading all installed software](https://www.ev3dev.org/docs/tutorials/upgrading-ev3dev/) should be enough to be up to date

Hopefully that is true :smile: but since there have been reports here of re-flashing the SD card making a difference, it is hard to say for sure what else could be going on.

**Comment by CubixSystem** [comment]
@dlech I try `ev3dev-stretch R2` and here is the result

1. After first time boot with new image Bluetooth was available in Brickman menu

2. After restart problem appear again
```
Oct 23 16:48:29 ev3dev hciattach[445]: Can't init device hci0: Operation not possible due to RF-kill (132)Can't initialize device: Operation not possible due to RF-kill
```

Full log:
```
robot@ev3dev:~$ ev3dev-sysinfo
Image file:         ev3dev-stretch-ev3-generic-2019-10-23
Kernel version:     4.14.117-ev3dev-2.3.4-ev3
Brickman:           0.10.2
BogoMIPS:           226.09
Bluetooth:
Board:              board0
BOARD_INFO_HW_REV=7
BOARD_INFO_MODEL=LEGO MINDSTORMS EV3
BOARD_INFO_ROM_REV=6
BOARD_INFO_SERIAL_NUM=0016534CDB22
BOARD_INFO_TYPE=main
robot@ev3dev:~$ sudo systemctl --no-pager status ev3-bluetooth -l
● ev3-bluetooth.service - LEGO MINDSTORMS EV3 built-in Bluetooth
   Loaded: loaded (/lib/systemd/system/ev3-bluetooth.service; static; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2019-10-23 16:48:29 UTC; 4min 18s ago
  Process: 445 ExecStart=/usr/bin/hciattach -t 30 /dev/ttyS2 texas 2000000 flow nosleep (code=exited, status=1/FAILURE)
  Process: 332 ExecStartPre=/bin/sleep 10 (code=exited, status=0/SUCCESS)

Oct 23 16:48:17 ev3dev systemd[1]: Starting LEGO MINDSTORMS EV3 built-in Bluetooth...
Oct 23 16:48:28 ev3dev hciattach[445]: Found a Texas Instruments' chip!
Oct 23 16:48:28 ev3dev hciattach[445]: Firmware file : /lib/firmware/ti-connectivity/TIInit_6.6.15.bts
Oct 23 16:48:28 ev3dev hciattach[445]: Loaded BTS script version 1
Oct 23 16:48:28 ev3dev hciattach[445]: texas: changing baud rate to 2000000, flow control to 1
Oct 23 16:48:29 ev3dev hciattach[445]: Can't init device hci0: Operation not possible due to RF-kill (132)Can't initialize device: Operation not possible due to RF-kill
Oct 23 16:48:29 ev3dev systemd[1]: ev3-bluetooth.service: Control process exited, code=exited status=1
Oct 23 16:48:29 ev3dev systemd[1]: Failed to start LEGO MINDSTORMS EV3 built-in Bluetooth.
Oct 23 16:48:29 ev3dev systemd[1]: ev3-bluetooth.service: Unit entered failed state.
Oct 23 16:48:29 ev3dev systemd[1]: ev3-bluetooth.service: Failed with result 'exit-code'.
```

3. After next restarts error in logs changed
```
robot@ev3dev:~$ sudo systemctl --no-pager status ev3-bluetooth -l
● ev3-bluetooth.service - LEGO MINDSTORMS EV3 built-in Bluetooth
   Loaded: loaded (/lib/systemd/system/ev3-bluetooth.service; static; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2019-11-25 22:40:49 UTC; 4min 2s ago
  Process: 439 ExecStart=/usr/bin/hciattach -t 30 /dev/ttyS2 texas 2000000 flow nosleep (code=exited, status=1/FAILURE)
  Process: 327 ExecStartPre=/bin/sleep 10 (code=exited, status=0/SUCCESS)

Nov 25 22:39:38 ev3dev systemd[1]: Starting LEGO MINDSTORMS EV3 built-in Bluetooth...
Nov 25 22:39:48 ev3dev hciattach[439]: Found a Texas Instruments' chip!
Nov 25 22:39:48 ev3dev hciattach[439]: Firmware file : /lib/firmware/ti-connectivity/TIInit_6.6.15.bts
Nov 25 22:39:48 ev3dev hciattach[439]: Loaded BTS script version 1
Nov 25 22:39:48 ev3dev hciattach[439]: texas: changing baud rate to 2000000, flow control to 1
Nov 25 22:40:18 ev3dev hciattach[439]: Initialization timed out.
Nov 25 22:40:49 ev3dev systemd[1]: ev3-bluetooth.service: Control process exited, code=exited status=1
Nov 25 22:40:49 ev3dev systemd[1]: Failed to start LEGO MINDSTORMS EV3 built-in Bluetooth.
Nov 25 22:40:49 ev3dev systemd[1]: ev3-bluetooth.service: Unit entered failed state.
Nov 25 22:40:49 ev3dev systemd[1]: ev3-bluetooth.service: Failed with result 'exit-code'.
```

**Comment by dlech** [comment]
Thanks for the detailed info. We know that `Initialization timed out.` is a timing issue and can possibly be improved by adjusting the delay.

If we can get the `Can't init device hci0: Operation not possible due to RF-kill` error again, please run `sudo journalctl --no-pager --boot=0` so we can see when Bluetooth was trying to start relative to other services.

**Comment by CubixSystem** [comment]
Caught it <details>
  <summary>Logs</summary>

  ```sh
robot@ev3dev:~$ sudo systemctl --no-pager status ev3-bluetooth -l
[sudo] password for robot:
● ev3-bluetooth.service - LEGO MINDSTORMS EV3 built-in Bluetooth
   Loaded: loaded (/lib/systemd/system/ev3-bluetooth.service; static; vendor preset: enabled)
   Active: failed (Result: exit-code) since Mon 2019-11-25 23:12:53 UTC; 1min 53s ago
  Process: 436 ExecStart=/usr/bin/hciattach -t 30 /dev/ttyS2 texas 2000000 flow nosleep (code=exited, status=1/FAILURE)
  Process: 359 ExecStartPre=/bin/sleep 10 (code=exited, status=0/SUCCESS)

Nov 25 23:12:42 ev3dev systemd[1]: Starting LEGO MINDSTORMS EV3 built-in Bluetooth...
Nov 25 23:12:52 ev3dev hciattach[436]: Found a Texas Instruments' chip!
Nov 25 23:12:52 ev3dev hciattach[436]: Firmware file : /lib/firmware/ti-connectivity/TIInit_6.6.15.bts
Nov 25 23:12:52 ev3dev hciattach[436]: Loaded BTS script version 1
Nov 25 23:12:52 ev3dev hciattach[436]: texas: changing baud rate to 2000000, flow control to 1
Nov 25 23:12:53 ev3dev hciattach[436]: Can't init device hci0: Operation not possible due to RF-kill (132)Can't initialize device: Operation not possible due to RF-kill
Nov 25 23:12:53 ev3dev systemd[1]: ev3-bluetooth.service: Control process exited, code=exited status=1
Nov 25 23:12:54 ev3dev systemd[1]: Failed to start LEGO MINDSTORMS EV3 built-in Bluetooth.
Nov 25 23:12:54 ev3dev systemd[1]: ev3-bluetooth.service: Unit entered failed state.
Nov 25 23:12:54 ev3dev systemd[1]: ev3-bluetooth.service: Failed with result 'exit-code'.
robot@ev3dev:~$ sudo journalctl --no-pager --boot=0
-- Logs begin at Thu 2016-11-03 17:16:45 UTC, end at Mon 2019-11-25 23:15:05 UTC. --
Nov 25 23:12:15 ev3dev kernel: Booting Linux on physical CPU 0x0
Nov 25 23:12:15 ev3dev kernel: Linux version 4.14.117-ev3dev-2.3.4-ev3 (david@freyr) (gcc version 6.4.1 20171012 (Linaro GCC 6.4-2017.11)) #1 PREEMPT Thu May 9 15:13:01 CDT 2019
Nov 25 23:12:15 ev3dev kernel: CPU: ARM926EJ-S [41069265] revision 5 (ARMv5TEJ), cr=0005317f
Nov 25 23:12:16 ev3dev kernel: CPU: VIVT data cache, VIVT instruction cache
Nov 25 23:12:16 ev3dev kernel: OF: fdt: Machine model: LEGO MINDSTORMS EV3
Nov 25 23:12:16 ev3dev kernel: Memory policy: Data cache writeback
Nov 25 23:12:16 ev3dev kernel: Reserved memory: created DMA memory pool at 0x80000000, size 0 MiB
Nov 25 23:12:16 ev3dev kernel: OF: reserved mem: initialized node serial@80000000, compatible id shared-dma-pool
Nov 25 23:12:16 ev3dev kernel: DaVinci da850/omap-l138/am18x variant 0x1
Nov 25 23:12:16 ev3dev kernel: On node 0 totalpages: 16384
Nov 25 23:12:16 ev3dev kernel: free_area_init_node: node 0, pgdat c07081e0, node_mem_map c3f79000
Nov 25 23:12:16 ev3dev kernel:   DMA zone: 128 pages used for memmap
Nov 25 23:12:16 ev3dev kernel:   DMA zone: 0 pages reserved
Nov 25 23:12:16 ev3dev kernel:   DMA zone: 16384 pages, LIFO batch:3
Nov 25 23:12:16 ev3dev kernel: pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
Nov 25 23:12:16 ev3dev kernel: pcpu-alloc: [0] 0
Nov 25 23:12:16 ev3dev kernel: Built 1 zonelists, mobility grouping on.  Total pages: 16256
Nov 25 23:12:16 ev3dev kernel: Kernel command line: console=tty1
Nov 25 23:12:16 ev3dev kernel: PID hash table entries: 256 (order: -2, 1024 bytes)
Nov 25 23:12:16 ev3dev kernel: Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
Nov 25 23:12:16 ev3dev kernel: Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
Nov 25 23:12:16 ev3dev kernel: Memory: 53248K/65536K available (5071K kernel code, 259K rwdata, 1660K rodata, 172K init, 158K bss, 12288K reserved, 0K cma-reserved)
Nov 25 23:12:16 ev3dev kernel: Virtual kernel memory layout:
                                   vector  : 0xffff0000 - 0xffff1000   (   4 kB)
                                   fixmap  : 0xffc00000 - 0xfff00000   (3072 kB)
                                   vmalloc : 0xc4800000 - 0xff800000   ( 944 MB)
                                   lowmem  : 0xc0000000 - 0xc4000000   (  64 MB)
                                   modules : 0xbf000000 - 0xc0000000   (  16 MB)
                                     .text : 0xc0008000 - 0xc04fc3c8   (5073 kB)
                                     .init : 0xc069d000 - 0xc06c8000   ( 172 kB)
                                     .data : 0xc06c8000 - 0xc0708ca0   ( 260 kB)
                                      .bss : 0xc070d854 - 0xc073542c   ( 159 kB)
Nov 25 23:12:16 ev3dev kernel: SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
Nov 25 23:12:16 ev3dev kernel: Preemptible hierarchical RCU implementation.
Nov 25 23:12:16 ev3dev kernel:         Tasks RCU enabled.
Nov 25 23:12:16 ev3dev kernel: NR_IRQS: 325
Nov 25 23:12:16 ev3dev kernel: clocksource: timer0_1: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 79635851949 ns
Nov 25 23:12:16 ev3dev kernel: sched_clock: 32 bits at 24MHz, resolution 41ns, wraps every 89478484971ns
Nov 25 23:12:16 ev3dev kernel: Console: colour dummy device 80x30
Nov 25 23:12:16 ev3dev kernel: console [tty1] enabled
Nov 25 23:12:16 ev3dev kernel: Calibrating delay loop... 227.32 BogoMIPS (lpj=1136640)
Nov 25 23:12:16 ev3dev kernel: pid_max: default: 32768 minimum: 301
Nov 25 23:12:16 ev3dev kernel: Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
Nov 25 23:12:16 ev3dev kernel: Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
Nov 25 23:12:16 ev3dev kernel: CPU: Testing write buffer coherency: ok
Nov 25 23:12:16 ev3dev kernel: Setting up static identity map for 0xc0008400 - 0xc000843c
Nov 25 23:12:16 ev3dev kernel: Hierarchical SRCU implementation.
Nov 25 23:12:16 ev3dev kernel: devtmpfs: initialized
Nov 25 23:12:16 ev3dev kernel: random: get_random_u32 called from bucket_table_alloc+0x14c/0x198 with crng_init=0
Nov 25 23:12:16 ev3dev kernel: clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
Nov 25 23:12:16 ev3dev kernel: futex hash table entries: 256 (order: -1, 3072 bytes)
Nov 25 23:12:16 ev3dev kernel: pinctrl core: initialized pinctrl subsystem
Nov 25 23:12:16 ev3dev kernel: NET: Registered protocol family 16
Nov 25 23:12:16 ev3dev kernel: DMA: preallocated 256 KiB pool for atomic coherent allocations
Nov 25 23:12:16 ev3dev kernel: cpuidle: using governor menu
Nov 25 23:12:16 ev3dev kernel: GPIO line 6 (batt_volt_en) hogged as output/high
Nov 25 23:12:16 ev3dev kernel: edma 1c00000.edma: memcpy is disabled
Nov 25 23:12:16 ev3dev kernel: edma 1c00000.edma: TI EDMA DMA engine driver
Nov 25 23:12:16 ev3dev kernel: edma 1e30000.edma: memcpy is disabled
Nov 25 23:12:16 ev3dev kernel: edma 1e30000.edma: TI EDMA DMA engine driver
Nov 25 23:12:16 ev3dev kernel: SCSI subsystem initialized
Nov 25 23:12:16 ev3dev kernel: usbcore: registered new interface driver usbfs
Nov 25 23:12:16 ev3dev kernel: usbcore: registered new interface driver hub
Nov 25 23:12:16 ev3dev kernel: usbcore: registered new device driver usb
Nov 25 23:12:16 ev3dev kernel: Advanced Linux Sound Architecture Driver Initialized.
Nov 25 23:12:16 ev3dev kernel: Bluetooth: Core ver 2.22
Nov 25 23:12:16 ev3dev kernel: NET: Registered protocol family 31
Nov 25 23:12:16 ev3dev kernel: Bluetooth: HCI device and connection manager initialized
Nov 25 23:12:16 ev3dev kernel: Bluetooth: HCI socket layer initialized
Nov 25 23:12:16 ev3dev kernel: Bluetooth: L2CAP socket layer initialized
Nov 25 23:12:16 ev3dev kernel: Bluetooth: SCO socket layer initialized
Nov 25 23:12:16 ev3dev kernel: clocksource: Switched to clocksource timer0_1
Nov 25 23:12:16 ev3dev kernel: NET: Registered protocol family 2
Nov 25 23:12:16 ev3dev kernel: TCP established hash table entries: 1024 (order: 0, 4096 bytes)
Nov 25 23:12:16 ev3dev kernel: TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
Nov 25 23:12:16 ev3dev kernel: TCP: Hash tables configured (established 1024 bind 1024)
Nov 25 23:12:16 ev3dev kernel: UDP hash table entries: 256 (order: 0, 4096 bytes)
Nov 25 23:12:16 ev3dev kernel: UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
Nov 25 23:12:16 ev3dev kernel: NET: Registered protocol family 1
Nov 25 23:12:16 ev3dev kernel: Unpacking initramfs...
Nov 25 23:12:16 ev3dev kernel: random: fast init done
Nov 25 23:12:16 ev3dev kernel: Freeing initrd memory: 4200K
Nov 25 23:12:17 ev3dev kernel: workingset: timestamp_bits=30 max_order=14 bucket_order=0
Nov 25 23:12:17 ev3dev kernel: io scheduler noop registered (default)
Nov 25 23:12:17 ev3dev kernel: io scheduler mq-deadline registered
Nov 25 23:12:17 ev3dev kernel: io scheduler kyber registered
Nov 25 23:12:17 ev3dev kernel: i2c-legoev3 i2c-legoev3.3: registered on input port 1
Nov 25 23:12:17 ev3dev kernel: i2c-legoev3 i2c-legoev3.4: registered on input port 2
Nov 25 23:12:17 ev3dev kernel: i2c-legoev3 i2c-legoev3.5: registered on input port 3
Nov 25 23:12:17 ev3dev kernel: i2c-legoev3 i2c-legoev3.6: registered on input port 4
Nov 25 23:12:17 ev3dev kernel: pinctrl-single 1c14120.pinmux: 160 pins at pa fec14120 size 80
Nov 25 23:12:17 ev3dev kernel: Serial: 8250/16550 driver, 3 ports, IRQ sharing disabled
Nov 25 23:12:17 ev3dev kernel: serial8250.0: ttyS0 at MMIO 0x1c42000 (irq = 25, base_baud = 14250000) is a TI DA8xx/66AK2x
Nov 25 23:12:17 ev3dev kernel: serial8250.1: ttyS1 at MMIO 0x1d0c000 (irq = 53, base_baud = 8250000) is a TI DA8xx/66AK2x
Nov 25 23:12:17 ev3dev kernel: serial8250.2: ttyS2 at MMIO 0x1d0d000 (irq = 61, base_baud = 8250000) is a TI DA8xx/66AK2x
Nov 25 23:12:17 ev3dev kernel: loop: module loaded
Nov 25 23:12:17 ev3dev kernel: zram: Added device: zram0
Nov 25 23:12:17 ev3dev kernel: spi_davinci spi_davinci.0: Controller at 0xfec41000
Nov 25 23:12:17 ev3dev kernel: [drm] Initialized st7586 1.0.0 20170801 for spi1.0 on minor 0
Nov 25 23:12:17 ev3dev kernel: Console: switching to colour frame buffer device 44x21
Nov 25 23:12:17 ev3dev kernel: st7586 spi1.0: fb0:  frame buffer device
Nov 25 23:12:17 ev3dev kernel: spi_davinci spi_davinci.1: Controller at 0xfef0e000
Nov 25 23:12:17 ev3dev kernel: ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
Nov 25 23:12:17 ev3dev kernel: ohci-da8xx: DA8XX
Nov 25 23:12:17 ev3dev kernel: usbcore: registered new interface driver usb-storage
Nov 25 23:12:17 ev3dev kernel: usb_phy_generic usb_phy_generic.0.auto: usb_phy_generic.0.auto supply vcc not found, using dummy regulator
Nov 25 23:12:17 ev3dev kernel: i2c /dev entries driver
Nov 25 23:12:17 ev3dev kernel: davinci-wdt davinci-wdt: heartbeat 60 sec
Nov 25 23:12:17 ev3dev kernel: Bluetooth: HCI UART driver ver 2.3
Nov 25 23:12:17 ev3dev kernel: Bluetooth: HCI UART protocol H4 registered
Nov 25 23:12:17 ev3dev kernel: Bluetooth: HCI UART protocol BCSP registered
Nov 25 23:12:17 ev3dev kernel: Bluetooth: HCI UART protocol LL registered
Nov 25 23:12:17 ev3dev kernel: davinci_mmc da830-mmc.0: Got CD GPIO
Nov 25 23:12:17 ev3dev kernel: davinci_mmc da830-mmc.0: Using DMA, 4-bit mode
Nov 25 23:12:17 ev3dev kernel: hidraw: raw HID events driver (C) Jiri Kosina
Nov 25 23:12:17 ev3dev kernel: usbcore: registered new interface driver usbhid
Nov 25 23:12:17 ev3dev kernel: usbhid: USB HID core driver
Nov 25 23:12:17 ev3dev kernel: usbcore: registered new interface driver r8712u
Nov 25 23:12:17 ev3dev kernel: NET: Registered protocol family 10
Nov 25 23:12:17 ev3dev kernel: Segment Routing with IPv6
Nov 25 23:12:17 ev3dev kernel: sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
Nov 25 23:12:17 ev3dev kernel: NET: Registered protocol family 17
Nov 25 23:12:17 ev3dev kernel: Bluetooth: BNEP (Ethernet Emulation) ver 1.3
Nov 25 23:12:17 ev3dev kernel: Bluetooth: BNEP socket layer initialized
Nov 25 23:12:17 ev3dev kernel: mmc0: host does not support reading read-only switch, assuming write-enable
Nov 25 23:12:18 ev3dev kernel: mmc0: new high speed SDHC card at address aaaa
Nov 25 23:12:18 ev3dev kernel: at24 0-0050: 16384 byte 24c128 EEPROM, read-only, 0 bytes/write
Nov 25 23:12:18 ev3dev kernel: mmcblk0: mmc0:aaaa SL32G 29.7 GiB
Nov 25 23:12:18 ev3dev kernel:  mmcblk0: p1 p2
Nov 25 23:12:18 ev3dev kernel: input: LEGO MINDSTORMS EV3 Speaker as /devices/platform/sound/input/input0
Nov 25 23:12:18 ev3dev kernel: remoteproc remoteproc0: pru1 is available
Nov 25 23:12:18 ev3dev kernel: adc ref: supplied by vcc5v
Nov 25 23:12:18 ev3dev kernel: ohci-da8xx ohci-da8xx: USB Host Controller
Nov 25 23:12:18 ev3dev kernel: ohci-da8xx ohci-da8xx: new USB bus registered, assigned bus number 1
Nov 25 23:12:18 ev3dev kernel: ohci-da8xx ohci-da8xx: irq 59, io mem 0x01e25000
Nov 25 23:12:18 ev3dev kernel: remoteproc remoteproc0: powering up pru1
Nov 25 23:12:18 ev3dev kernel: remoteproc remoteproc0: Booting fw image ev3-pru1.fw, size 82036
Nov 25 23:12:18 ev3dev kernel: virtio_rpmsg_bus virtio0: creating channel ev3-tacho-rpmsg addr 0x0
Nov 25 23:12:18 ev3dev kernel: virtio_rpmsg_bus virtio0: rpmsg host is online
Nov 25 23:12:18 ev3dev kernel: remoteproc remoteproc0: registered virtio0 (type 7)
Nov 25 23:12:18 ev3dev kernel: remoteproc remoteproc0: remote processor pru1 is now up
Nov 25 23:12:18 ev3dev kernel: ev3_pru virtio0.ev3-tacho-rpmsg.-1.0: new channel: 0x400 -> 0x0!
Nov 25 23:12:18 ev3dev kernel: usb usb1: New USB device found, idVendor=1d6b, idProduct=0001
Nov 25 23:12:18 ev3dev kernel: usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
Nov 25 23:12:18 ev3dev kernel: usb usb1: Product: USB Host Controller
Nov 25 23:12:18 ev3dev kernel: usb usb1: Manufacturer: Linux 4.14.117-ev3dev-2.3.4-ev3 ohci_hcd
Nov 25 23:12:18 ev3dev kernel: usb usb1: SerialNumber: ohci-da8xx
Nov 25 23:12:18 ev3dev kernel: hub 1-0:1.0: USB hub found
Nov 25 23:12:18 ev3dev kernel: hub 1-0:1.0: 1 port detected
Nov 25 23:12:18 ev3dev kernel: lego-port port0: Registered 'ev3-ports:in1' on 'ev3-ports:in1'.
Nov 25 23:12:18 ev3dev kernel: lego-port port1: Registered 'ev3-ports:in2' on 'ev3-ports:in2'.
Nov 25 23:12:18 ev3dev kernel: lego-port port2: Registered 'ev3-ports:in3' on 'ev3-ports:in3'.
Nov 25 23:12:18 ev3dev kernel: lego-port port3: Registered 'ev3-ports:in4' on 'ev3-ports:in4'.
Nov 25 23:12:18 ev3dev kernel: lego-port port4: Registered 'ev3-ports:outA' on 'ev3-ports:outA'.
Nov 25 23:12:18 ev3dev kernel: lego-port port5: Registered 'ev3-ports:outB' on 'ev3-ports:outB'.
Nov 25 23:12:19 ev3dev kernel: lego-port port6: Registered 'ev3-ports:outC' on 'ev3-ports:outC'.
Nov 25 23:12:19 ev3dev kernel: lego-port port7: Registered 'ev3-ports:outD' on 'ev3-ports:outD'.
Nov 25 23:12:19 ev3dev kernel: ti_omapl_pru_suart 1c30000.serial: assigned reserved memory node serial@80000000
Nov 25 23:12:19 ev3dev kernel: 1c30000.serial: ttySU0 at MMIO 0x1d00000 (irq = 254, base_baud = 8250000) is a suart_tty
Nov 25 23:12:19 ev3dev kernel: 1c30000.serial: ttySU1 at MMIO 0x1d00000 (irq = 256, base_baud = 8250000) is a suart_tty
Nov 25 23:12:19 ev3dev kernel: ti_omapl_pru_suart 1c30000.serial: ti_omapl_pru_suart device registered (pru_clk=228000000, asp_clk=132000000)
Nov 25 23:12:19 ev3dev kernel: input: EV3 Brick Buttons as /devices/platform/gpio_keys/input/input1
Nov 25 23:12:19 ev3dev kernel: ALSA device list:
Nov 25 23:12:19 ev3dev kernel:   #0: LEGO MINDSTORMS EV3 Speaker connected to ehrpwm.0
Nov 25 23:12:19 ev3dev kernel: Freeing unused kernel memory: 172K
Nov 25 23:12:19 ev3dev kernel: This architecture does not have kernel memory protection.
Nov 25 23:12:19 ev3dev kernel: random: systemd-udevd: uninitialized urandom read (16 bytes read)
Nov 25 23:12:19 ev3dev kernel: random: systemd-udevd: uninitialized urandom read (16 bytes read)
Nov 25 23:12:19 ev3dev kernel: random: udevadm: uninitialized urandom read (16 bytes read)
Nov 25 23:12:19 ev3dev kernel: EXT4-fs (mmcblk0p2): mounted filesystem with ordered data mode. Opts: (null)
Nov 25 23:12:19 ev3dev systemd[1]: System time before build time, advancing clock.
Nov 25 23:12:19 ev3dev kernel: ip_tables: (C) 2000-2006 Netfilter Core Team
Nov 25 23:12:19 ev3dev systemd[1]: systemd 232 running in system mode. (+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN)
Nov 25 23:12:19 ev3dev systemd[1]: Detected architecture arm.
Nov 25 23:12:19 ev3dev systemd[1]: Set hostname to <ev3dev>.
Nov 25 23:12:19 ev3dev kernel: random: crng init done
Nov 25 23:12:19 ev3dev kernel: random: 7 urandom warning(s) missed due to ratelimiting
Nov 25 23:12:19 ev3dev systemd[1]: openrobertalab.service: Cannot add dependency job, ignoring: Unit openrobertalab.service is masked.
Nov 25 23:12:19 ev3dev systemd[1]: Created slice User and Session Slice.
Nov 25 23:12:19 ev3dev systemd[1]: Listening on fsck to fsckd communication Socket.
Nov 25 23:12:19 ev3dev systemd[1]: Created slice System Slice.
Nov 25 23:12:19 ev3dev systemd[1]: Mounting RPC Pipe File System...
Nov 25 23:12:19 ev3dev systemd[1]: Created slice system-systemd\x2dfsck.slice.
Nov 25 23:12:19 ev3dev systemd[1]: Reached target Slices.
Nov 25 23:12:19 ev3dev kernel: RPC: Registered named UNIX socket transport module.
Nov 25 23:12:19 ev3dev kernel: RPC: Registered udp transport module.
Nov 25 23:12:19 ev3dev kernel: RPC: Registered tcp transport module.
Nov 25 23:12:19 ev3dev kernel: RPC: Registered tcp NFSv4.1 backchannel transport module.
Nov 25 23:12:19 ev3dev kernel: zram0: detected capacity change from 0 to 100663296
Nov 25 23:12:19 ev3dev kernel: EXT4-fs (mmcblk0p2): re-mounted. Opts: errors=remount-ro
Nov 25 23:12:19 ev3dev kernel: Adding 98300k swap on /dev/zram0.  Priority:16383 extents:1 across:98300k SS
Nov 25 23:12:19 ev3dev systemd[1]: Mounting Configuration File System...
Nov 25 23:12:19 ev3dev systemd[1]: Starting Load/Save Random Seed...
Nov 25 23:12:19 ev3dev systemd[1]: Starting udev Coldplug all Devices...
Nov 25 23:12:19 ev3dev systemd[1]: Mounted Configuration File System.
Nov 25 23:12:19 ev3dev systemd[1]: Started swap with zram.
Nov 25 23:12:19 ev3dev systemd[1]: Started Create Static Device Nodes in /dev.
Nov 25 23:12:19 ev3dev systemd[1]: Started Apply Kernel Variables.
Nov 25 23:12:19 ev3dev systemd[1]: Started Load/Save Random Seed.
Nov 25 23:12:19 ev3dev systemd[1]: Starting udev Kernel Device Manager...
Nov 25 23:12:19 ev3dev systemd[1]: Reached target Swap.
Nov 25 23:12:19 ev3dev systemd-journald[157]: Journal started
Nov 25 23:12:19 ev3dev systemd-journald[157]: Runtime journal (/run/log/journal/cb4688e460244c70bbcea7c423e91117) is 512.0K, max 1.0M, 512.0K free.
Nov 25 23:12:16 ev3dev systemd-modules-load[162]: Inserted module 'libcomposite'
Nov 25 23:12:16 ev3dev systemd-modules-load[162]: Inserted module 'uinput'
Nov 25 23:12:19 ev3dev systemd[1]: Starting Flush Journal to Persistent Storage...
Nov 25 23:12:19 ev3dev fake-hwclock[137]: Mon Nov 25 23:12:15 UTC 2019
Nov 25 23:12:20 ev3dev systemd[1]: Started udev Kernel Device Manager.
Nov 25 23:12:20 ev3dev systemd-journald[157]: Time spent on flushing to /var is 3.332202s for 199 entries.
Nov 25 23:12:20 ev3dev systemd-journald[157]: System journal (/var/log/journal/cb4688e460244c70bbcea7c423e91117) is 56.0M, max 2.9G, 2.8G free.
Nov 25 23:12:22 ev3dev systemd[1]: dev-hugepages.mount: Couldn't determine result for ConditionVirtualization=!private-users, assuming failed: No such file or directory
Nov 25 23:12:25 ev3dev systemd[1]: Started Flush Journal to Persistent Storage.
Nov 25 23:12:25 ev3dev systemd[1]: Started Set the console keyboard layout.
Nov 25 23:12:25 ev3dev systemd[1]: Reached target Local File Systems (Pre).
Nov 25 23:12:27 ev3dev systemd[1]: Started udev Coldplug all Devices.
Nov 25 23:12:27 ev3dev systemd[1]: Starting udev Wait for Complete Device Initialization...
Nov 25 23:12:28 ev3dev systemd[1]: Found device /sys/devices/platform/legoev3-bluetooth.
Nov 25 23:12:34 ev3dev systemd[1]: Found device /dev/mmcblk0p1.
Nov 25 23:12:35 ev3dev systemd[1]: Listening on Load/Save RF Kill Switch Status /dev/rfkill Watch.
Nov 25 23:12:35 ev3dev systemd[1]: Reached target Sound Card.
Nov 25 23:12:35 ev3dev systemd[1]: dev-hugepages.mount: Couldn't determine result for ConditionVirtualization=!private-users, assuming failed: No such file or directory
Nov 25 23:12:35 ev3dev systemd[1]: Created slice system-ev3\x2dusb.slice.
Nov 25 23:12:35 ev3dev systemd[1]: Starting File System Check on /dev/mmcblk0p1...
Nov 25 23:12:36 ev3dev systemd[1]: Started File System Check Daemon to report status.
Nov 25 23:12:37 ev3dev systemd-fsck[257]: fsck.fat 4.1 (2017-01-24)
Nov 25 23:12:37 ev3dev systemd-fsck[257]: /dev/mmcblk0p1: 8 files, 116/24519 clusters
Nov 25 23:12:37 ev3dev systemd[1]: Started File System Check on /dev/mmcblk0p1.
Nov 25 23:12:37 ev3dev systemd[1]: Mounting /boot/flash...
Nov 25 23:12:37 ev3dev systemd[1]: Started udev Wait for Complete Device Initialization.
Nov 25 23:12:37 ev3dev systemd[1]: Mounted /boot/flash.
Nov 25 23:12:37 ev3dev systemd[1]: Reached target Local File Systems.
Nov 25 23:12:38 ev3dev systemd[1]: Starting Raise network interfaces...
Nov 25 23:12:38 ev3dev systemd[1]: Starting Preprocess NFS configuration...
Nov 25 23:12:38 ev3dev systemd[1]: Starting Set console font and keymap...
Nov 25 23:12:38 ev3dev systemd[1]: Starting Create Volatile Files and Directories...
Nov 25 23:12:38 ev3dev systemd[1]: Started Preprocess NFS configuration.
Nov 25 23:12:38 ev3dev systemd[1]: Started Set console font and keymap.
Nov 25 23:12:38 ev3dev systemd[1]: Reached target NFS client services.
Nov 25 23:12:38 ev3dev systemd[1]: Reached target Remote File Systems (Pre).
Nov 25 23:12:38 ev3dev systemd[1]: Reached target Remote File Systems.
Nov 25 23:12:39 ev3dev systemd-tmpfiles[290]: Cannot set file attribute for '/var/log/journal', value=0x00800000, mask=0x00800000: Operation not supported
Nov 25 23:12:39 ev3dev systemd-tmpfiles[290]: Cannot set file attribute for '/var/log/journal/cb4688e460244c70bbcea7c423e91117', value=0x00800000, mask=0x00800000: Operation not supported
Nov 25 23:12:39 ev3dev systemd[1]: Started Create Volatile Files and Directories.
Nov 25 23:12:39 ev3dev systemd[1]: Reached target System Time Synchronized.
Nov 25 23:12:39 ev3dev systemd[1]: Starting Update UTMP about System Boot/Shutdown...
Nov 25 23:12:40 ev3dev systemd[1]: Started Update UTMP about System Boot/Shutdown.
Nov 25 23:12:40 ev3dev systemd[1]: Reached target System Initialization.
Nov 25 23:12:40 ev3dev systemd[1]: Started Daily Cleanup of Temporary Directories.
Nov 25 23:12:40 ev3dev systemd[1]: Reached target Timers.
Nov 25 23:12:40 ev3dev systemd[1]: Listening on Avahi mDNS/DNS-SD Stack Activation Socket.
Nov 25 23:12:40 ev3dev systemd[1]: Listening on D-Bus System Message Bus Socket.
Nov 25 23:12:40 ev3dev systemd[1]: Reached target Sockets.
Nov 25 23:12:40 ev3dev systemd[1]: Reached target Basic System.
Nov 25 23:12:40 ev3dev systemd[1]: Starting Avahi mDNS/DNS-SD Stack...
Nov 25 23:12:40 ev3dev systemd[1]: Reached target Login Prompts.
Nov 25 23:12:40 ev3dev systemd[1]: Starting Login Service...
Nov 25 23:12:40 ev3dev systemd[1]: Starting Console runner for robot...
Nov 25 23:12:40 ev3dev systemd[1]: Starting USB Gadget for LEGO MINDSTORMS EV3 hardware...
Nov 25 23:12:40 ev3dev systemd[1]: Started D-Bus System Message Bus.
Nov 25 23:12:41 ev3dev avahi-daemon[322]: Found user 'avahi' (UID 107) and group 'avahi' (GID 113).
Nov 25 23:12:41 ev3dev avahi-daemon[322]: Successfully dropped root privileges.
Nov 25 23:12:41 ev3dev avahi-daemon[322]: avahi-daemon 0.6.32 starting up.
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Successfully called chroot().
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Successfully dropped remaining capabilities.
Nov 25 23:12:42 ev3dev ev3-usb.sh[325]: Setting up gadget...
Nov 25 23:12:42 ev3dev avahi-daemon[334]: chroot.c: open() failed: No such file or directory
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Failed to open /etc/resolv.conf: Invalid argument
Nov 25 23:12:42 ev3dev systemd[1]: Started Avahi mDNS/DNS-SD Stack.
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Loading service file /services/sftp-ssh.service.
Nov 25 23:12:42 ev3dev systemd[1]: Started Brick Daemon.
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Loading service file /services/ssh.service.
Nov 25 23:12:42 ev3dev systemd[1]: Starting LEGO MINDSTORMS EV3 built-in Bluetooth...
Nov 25 23:12:42 ev3dev dbus[326]: [system] Successfully activated service 'org.freedesktop.systemd1'
Nov 25 23:12:42 ev3dev systemd[1]: Starting Save/Restore Sound Card State...
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Network interface enumeration completed.
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Server startup complete. Host name is ev3dev.local. Local service cookie is 3430464546.
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Service "ev3dev" (/services/ssh.service) successfully established.
Nov 25 23:12:42 ev3dev avahi-daemon[322]: Service "ev3dev" (/services/sftp-ssh.service) successfully established.
Nov 25 23:12:43 ev3dev systemd[1]: Started Console runner for robot.
Nov 25 23:12:43 ev3dev kernel: using random self ethernet address
Nov 25 23:12:43 ev3dev kernel: using random host ethernet address
Nov 25 23:12:43 ev3dev systemd[1]: Started Save/Restore Sound Card State.
Nov 25 23:12:43 ev3dev kernel: using random self ethernet address
Nov 25 23:12:43 ev3dev kernel: using random host ethernet address
Nov 25 23:12:44 ev3dev kernel: usb0: HOST MAC 12:16:53:4c:db:22
Nov 25 23:12:44 ev3dev kernel: usb0: MAC 02:16:53:4c:db:22
Nov 25 23:12:44 ev3dev kernel: usb1: HOST MAC 32:16:53:4c:db:22
Nov 25 23:12:44 ev3dev kernel: usb1: MAC 22:16:53:4c:db:22
Nov 25 23:12:44 ev3dev ev3-usb.sh[325]: Done.
Nov 25 23:12:44 ev3dev systemd[1]: Started USB Gadget for LEGO MINDSTORMS EV3 hardware.
Nov 25 23:12:44 ev3dev systemd-logind[323]: New seat seat0.
Nov 25 23:12:44 ev3dev systemd[1]: Started Login Service.
Nov 25 23:12:45 ev3dev kernel: configfs-gadget gadget: high-speed config #1: c
Nov 25 23:12:46 ev3dev systemd[1]: Started Raise network interfaces.
Nov 25 23:12:52 ev3dev hciattach[436]: Found a Texas Instruments' chip!
Nov 25 23:12:52 ev3dev hciattach[436]: Firmware file : /lib/firmware/ti-connectivity/TIInit_6.6.15.bts
Nov 25 23:12:52 ev3dev hciattach[436]: Loaded BTS script version 1
Nov 25 23:12:52 ev3dev hciattach[436]: texas: changing baud rate to 2000000, flow control to 1
Nov 25 23:12:53 ev3dev systemd[1]: Starting Load/Save RF Kill Switch Status...
Nov 25 23:12:53 ev3dev systemd[1]: Started Load/Save RF Kill Switch Status.
Nov 25 23:12:53 ev3dev hciattach[436]: Can't init device hci0: Operation not possible due to RF-kill (132)Can't initialize device: Operation not possible due to RF-kill
Nov 25 23:12:53 ev3dev systemd[1]: ev3-bluetooth.service: Control process exited, code=exited status=1
Nov 25 23:12:54 ev3dev systemd[1]: Failed to start LEGO MINDSTORMS EV3 built-in Bluetooth.
Nov 25 23:12:54 ev3dev systemd[1]: ev3-bluetooth.service: Unit entered failed state.
Nov 25 23:12:54 ev3dev systemd[1]: ev3-bluetooth.service: Failed with result 'exit-code'.
Nov 25 23:12:54 ev3dev systemd[1]: Starting Connection service...
Nov 25 23:12:54 ev3dev systemd[1]: Starting Bluetooth service...
Nov 25 23:12:54 ev3dev bluetoothd[446]: Bluetooth daemon 5.43
Nov 25 23:12:54 ev3dev systemd[1]: Started Bluetooth service.
Nov 25 23:12:54 ev3dev bluetoothd[446]: Starting SDP server
Nov 25 23:12:54 ev3dev systemd[1]: Reached target Bluetooth.
Nov 25 23:12:55 ev3dev systemd[1]: bluetooth.target: Unit not needed anymore. Stopping.
Nov 25 23:12:55 ev3dev systemd[1]: Stopped target Bluetooth.
Nov 25 23:12:55 ev3dev dbus[326]: [system] Activating via systemd: service name='org.freedesktop.hostname1' unit='dbus-org.freedesktop.hostname1.service'
Nov 25 23:12:55 ev3dev bluetoothd[446]: Bluetooth management interface 1.14 initialized
Nov 25 23:12:55 ev3dev connmand[445]: Connection Manager version 1.36
Nov 25 23:12:55 ev3dev systemd[1]: Started Connection service.
Nov 25 23:12:55 ev3dev systemd[1]: Starting Wait for network to be configured by ConnMan...
Nov 25 23:12:55 ev3dev systemd[1]: Reached target Network.
Nov 25 23:12:55 ev3dev systemd[1]: Starting Network Name Resolution...
Nov 25 23:12:55 ev3dev connmand[445]: iptables support missing error 2 (No such file or directory)
Nov 25 23:12:55 ev3dev connmand[445]: Failed to flush table 'filter': Invalid argument
Nov 25 23:12:55 ev3dev systemd[1]: Starting OpenBSD Secure Shell server...
Nov 25 23:12:55 ev3dev systemd[1]: Starting Permit User Sessions...
Nov 25 23:12:56 ev3dev systemd[1]: Starting Hostname Service...
Nov 25 23:12:56 ev3dev connmand[445]: iptables support missing error 2 (No such file or directory)
Nov 25 23:12:56 ev3dev connmand[445]: Failed to flush table 'mangle': Invalid argument
Nov 25 23:12:56 ev3dev systemd[1]: Started Permit User Sessions.
Nov 25 23:12:56 ev3dev kernel: nf_conntrack version 0.5.0 (1024 buckets, 4096 max)
Nov 25 23:12:56 ev3dev systemd[1]: Started Brick Manager.
Nov 25 23:12:57 ev3dev connmand[445]: Checking loopback interface settings
Nov 25 23:12:57 ev3dev connmand[445]: System hostname is ev3dev
Nov 25 23:12:58 ev3dev systemd-resolved[450]: Positive Trust Anchors:
Nov 25 23:12:58 ev3dev systemd-resolved[450]: . IN DS 19036 8 2 49aac11d7b6f6446702e54a1607371607a1a41855200fd2ce1cdde32f24e8fb5
Nov 25 23:12:58 ev3dev systemd-resolved[450]: . IN DS 20326 8 2 e06d44b80b8f1d39a95c0b0d7c65d08458e880409bbc683457104237c7f8ec8d
Nov 25 23:12:58 ev3dev systemd-resolved[450]: Negative trust anchors: 10.in-addr.arpa 16.172.in-addr.arpa 17.172.in-addr.arpa 18.172.in-addr.arpa 19.172.in-addr.arpa 20.172.in-addr.arpa 21.172.in-addr.arpa 22.172.in-addr.arpa 23.172.in-addr.arpa 24.172.in-addr.arpa 25.172.in-addr.arpa 26.172.in-addr.arpa 27.172.in-addr.arpa 28.172.in-addr.arpa 29.172.in-addr.arpa 30.172.in-addr.arpa 31.172.in-addr.arpa 168.192.in-addr.arpa d.f.ip6.arpa corp home internal intranet lan local private test
Nov 25 23:12:58 ev3dev systemd-resolved[450]: Using system hostname 'ev3dev'.
Nov 25 23:12:59 ev3dev dbus[326]: [system] Activating via systemd: service name='fi.w1.wpa_supplicant1' unit='wpa_supplicant.service'
Nov 25 23:12:59 ev3dev dbus[326]: [system] Successfully activated service 'org.freedesktop.hostname1'
Nov 25 23:12:59 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:12:59 ev3dev connmand[445]: lo {newlink} index 1 address 00:00:00:00:00:00 mtu 65536
Nov 25 23:12:59 ev3dev connmand[445]: lo {newlink} index 1 operstate 0 <UNKNOWN>
Nov 25 23:12:59 ev3dev connmand[445]: sit0 {newlink} index 2 address 00:00:00:00:08:00 mtu 1480
Nov 25 23:12:59 ev3dev connmand[445]: sit0 {newlink} index 2 operstate 2 <DOWN>
Nov 25 23:12:59 ev3dev connmand[445]: usb0 {create} index 3 type 1 <ETHER>
Nov 25 23:13:00 ev3dev kernel: IPv6: ADDRCONF(NETDEV_UP): usb1: link is not ready
Nov 25 23:12:59 ev3dev connmand[445]: usb0 {update} flags 4098 <DOWN>
Nov 25 23:13:00 ev3dev systemd[1]: Started Hostname Service.
Nov 25 23:12:59 ev3dev connmand[445]: usb0 {newlink} index 3 address 02:16:53:4C:DB:22 mtu 1500
Nov 25 23:13:00 ev3dev systemd[1]: Starting WPA supplicant...
Nov 25 23:12:59 ev3dev connmand[445]: usb0 {newlink} index 3 operstate 2 <DOWN>
Nov 25 23:12:59 ev3dev connmand[445]: Adding interface usb0 [ gadget ]
Nov 25 23:12:59 ev3dev connmand[445]: usb1 {create} index 4 type 1 <ETHER>
Nov 25 23:12:59 ev3dev connmand[445]: usb1 {update} flags 4098 <DOWN>
Nov 25 23:12:59 ev3dev connmand[445]: usb1 {newlink} index 4 address 22:16:53:4C:DB:22 mtu 1500
Nov 25 23:12:59 ev3dev connmand[445]: usb1 {newlink} index 4 operstate 2 <DOWN>
Nov 25 23:13:00 ev3dev connmand[445]: Adding interface usb1 [ gadget ]
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {update} flags 102467 <UP,RUNNING,LOWER_UP>
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {newlink} index 3 address 02:16:53:4C:DB:22 mtu 1500
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {newlink} index 3 operstate 6 <UP>
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route ff00:: gw :: scope 0 <UNIVERSE>
Nov 25 23:13:00 ev3dev avahi-daemon[322]: Joining mDNS multicast group on interface usb0.IPv4 with address 192.168.2.3.
Nov 25 23:13:01 ev3dev systemd[1]: Started Wait for network to be configured by ConnMan.
Nov 25 23:13:00 ev3dev avahi-daemon[322]: New relevant interface usb0.IPv4 for mDNS.
Nov 25 23:13:01 ev3dev systemd[1]: Reached target Network is Online.
Nov 25 23:13:00 ev3dev avahi-daemon[322]: Registering new address record for 192.168.2.3 on usb0.IPv4.
Nov 25 23:13:01 ev3dev systemd[1]: Starting LSB: Start NTP daemon...
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route fe80:: gw :: scope 0 <UNIVERSE>
Nov 25 23:13:01 ev3dev systemd[1]: Started OpenBSD Secure Shell server.
Nov 25 23:13:00 ev3dev connmand[445]: usb1 {update} flags 36867 <UP>
Nov 25 23:13:00 ev3dev connmand[445]: usb1 {newlink} index 4 address 22:16:53:4C:DB:22 mtu 1500
Nov 25 23:13:00 ev3dev connmand[445]: usb1 {newlink} index 4 operstate 2 <DOWN>
Nov 25 23:13:02 ev3dev systemd[1]: Started WPA supplicant.
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} address 192.168.2.3/24 label usb0 family 2
Nov 25 23:13:02 ev3dev systemd[1]: Started Network Name Resolution.
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route 192.168.2.0 gw 0.0.0.0 scope 253 <LINK>
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route 192.168.2.1 gw 0.0.0.0 scope 253 <LINK>
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route 0.0.0.0 gw 192.168.2.1 scope 0 <UNIVERSE>
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route 0.0.0.0 gw 192.168.2.1 scope 0 <UNIVERSE>
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route 192.168.2.0 gw 0.0.0.0 scope 253 <LINK>
Nov 25 23:13:00 ev3dev connmand[445]: usb0 {add} route 192.168.2.1 gw 0.0.0.0 scope 253 <LINK>
Nov 25 23:13:00 ev3dev connmand[445]: The name net.connman.vpn was not provided by any .service files
Nov 25 23:13:01 ev3dev avahi-daemon[322]: Joining mDNS multicast group on interface usb0.IPv6 with address fe80::16:53ff:fe4c:db22.
Nov 25 23:13:01 ev3dev avahi-daemon[322]: New relevant interface usb0.IPv6 for mDNS.
Nov 25 23:13:01 ev3dev avahi-daemon[322]: Registering new address record for fe80::16:53ff:fe4c:db22 on usb0.*.
Nov 25 23:13:01 ev3dev sshd[470]: Server listening on 0.0.0.0 port 22.
Nov 25 23:13:01 ev3dev sshd[470]: Server listening on :: port 22.
Nov 25 23:13:01 ev3dev dbus[326]: [system] Successfully activated service 'fi.w1.wpa_supplicant1'
Nov 25 23:13:02 ev3dev wpa_supplicant[473]: Successfully initialized wpa_supplicant
Nov 25 23:13:03 ev3dev ntpd[483]: ntpd 4.2.8p10@1.3728-o Sun Feb 25 21:42:38 UTC 2018 (1): Starting
Nov 25 23:13:03 ev3dev ntpd[483]: Command line: /usr/sbin/ntpd -p /var/run/ntpd.pid -g -u 105:108
Nov 25 23:13:03 ev3dev ntp[475]: Starting NTP server: ntpd.
Nov 25 23:13:03 ev3dev systemd[1]: Started LSB: Start NTP daemon.
Nov 25 23:13:03 ev3dev systemd[1]: Reached target Multi-User System.
Nov 25 23:13:03 ev3dev systemd[1]: Starting LEGO MINDSTORMS EV3 LEDs...
Nov 25 23:13:03 ev3dev systemd[1]: Reached target Graphical Interface.
Nov 25 23:13:03 ev3dev systemd[1]: Starting Update UTMP about System Runlevel Changes...
Nov 25 23:13:03 ev3dev ntpd[486]: proto: precision = 8.875 usec (-17)
Nov 25 23:13:04 ev3dev ntpd[486]: Listen and drop on 0 v6wildcard [::]:123
Nov 25 23:13:04 ev3dev ntpd[486]: Listen and drop on 1 v4wildcard 0.0.0.0:123
Nov 25 23:13:04 ev3dev systemd[1]: Started LEGO MINDSTORMS EV3 LEDs.
Nov 25 23:13:04 ev3dev ntpd[486]: Listen normally on 2 lo 127.0.0.1:123
Nov 25 23:13:04 ev3dev ntpd[486]: Listen normally on 3 usb0 192.168.2.3:123
Nov 25 23:13:04 ev3dev ntpd[486]: Listen normally on 4 lo [::1]:123
Nov 25 23:13:04 ev3dev ntpd[486]: Listen normally on 5 usb0 [fe80::16:53ff:fe4c:db22%3]:123
Nov 25 23:13:04 ev3dev ntpd[486]: Listening on routing socket on fd #22 for interface updates
Nov 25 23:13:05 ev3dev systemd[1]: Started Update UTMP about System Runlevel Changes.
Nov 25 23:13:06 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:13:07 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:13:08 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:13:09 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:13:11 ev3dev connmand[445]: IPv4 online check failed for 0x5afe70 Wired
Nov 25 23:13:12 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:13:14 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:13:14 ev3dev systemd-resolved[450]: Using degraded feature set (UDP+EDNS0+DO) for DNS server ::1.
Nov 25 23:13:18 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:13:18 ev3dev systemd-resolved[450]: Using degraded feature set (UDP+EDNS0+DO) for DNS server 127.0.0.1.
Nov 25 23:13:22 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:13:36 ev3dev ntpd[486]: error resolving pool 0.debian.pool.ntp.org: Name or service not known (-2)
Nov 25 23:13:41 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:13:46 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:13:51 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:14:02 ev3dev systemd[1]: sys-subsystem-udc-devices-musb-hdrc.1.auto.device: Job sys-subsystem-udc-devices-musb-hdrc.1.auto.device/start timed out.
Nov 25 23:14:02 ev3dev systemd[1]: Timed out waiting for device sys-subsystem-udc-devices-musb-hdrc.1.auto.device.
Nov 25 23:14:02 ev3dev systemd[1]: sys-subsystem-udc-devices-musb-hdrc.1.auto.device: Job sys-subsystem-udc-devices-musb-hdrc.1.auto.device/start failed with result 'timeout'.
Nov 25 23:14:02 ev3dev systemd[1]: Startup finished in 9.731s (kernel) + 1min 51.026s (userspace) = 2min 757ms.
Nov 25 23:14:06 ev3dev ntpd[486]: error resolving pool 1.debian.pool.ntp.org: Name or service not known (-2)
Nov 25 23:14:11 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:14:11 ev3dev systemd-resolved[450]: Using degraded feature set (UDP+EDNS0) for DNS server ::1.
Nov 25 23:14:16 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:14:16 ev3dev systemd-resolved[450]: Using degraded feature set (UDP+EDNS0) for DNS server 127.0.0.1.
Nov 25 23:14:32 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:14:36 ev3dev ntpd[486]: error resolving pool 2.debian.pool.ntp.org: Name or service not known (-2)
Nov 25 23:14:37 ev3dev sshd[532]: Accepted keyboard-interactive/pam for robot from 192.168.2.1 port 52145 ssh2
Nov 25 23:14:37 ev3dev sshd[532]: pam_unix(sshd:session): session opened for user robot by (uid=0)
Nov 25 23:14:37 ev3dev systemd[1]: Created slice User Slice of robot.
Nov 25 23:14:37 ev3dev systemd[1]: Starting User Manager for UID 1000...
Nov 25 23:14:38 ev3dev systemd[1]: Started Session c1 of user robot.
Nov 25 23:14:38 ev3dev systemd-logind[323]: New session c1 of user robot.
Nov 25 23:14:38 ev3dev systemd[539]: pam_unix(systemd-user:session): session opened for user robot by (uid=0)
Nov 25 23:14:38 ev3dev systemd[539]: Reached target Paths.
Nov 25 23:14:38 ev3dev systemd[539]: Reached target Sockets.
Nov 25 23:14:38 ev3dev systemd[539]: Reached target Timers.
Nov 25 23:14:38 ev3dev systemd[539]: Reached target Basic System.
Nov 25 23:14:38 ev3dev systemd[539]: Reached target Default.
Nov 25 23:14:38 ev3dev systemd[539]: Startup finished in 442ms.
Nov 25 23:14:38 ev3dev systemd[1]: Started User Manager for UID 1000.
Nov 25 23:14:41 ev3dev systemd-resolved[450]: Switching to system DNS server 127.0.0.1.
Nov 25 23:14:47 ev3dev sudo[557]:    robot : TTY=pts/0 ; PWD=/home/robot ; USER=root ; COMMAND=/bin/systemctl --no-pager status ev3-bluetooth -l
Nov 25 23:14:47 ev3dev sudo[557]: pam_unix(sudo:session): session opened for user root by robot(uid=0)
Nov 25 23:14:48 ev3dev sudo[557]: pam_unix(sudo:session): session closed for user root
Nov 25 23:14:57 ev3dev systemd-resolved[450]: Switching to system DNS server ::1.
Nov 25 23:15:05 ev3dev sudo[567]:    robot : TTY=pts/0 ; PWD=/home/robot ; USER=root ; COMMAND=/bin/journalctl --no-pager --boot=0
Nov 25 23:15:05 ev3dev sudo[567]: pam_unix(sudo:session): session opened for user root by robot(uid=0)
Nov 25 23:15:06 ev3dev ntpd[486]: error resolving pool 3.debian.pool.ntp.org: Name or service not known (-2)
  ```
  
</details>

**Comment by dlech** [comment]
Thanks again for the logs.

I don't know why I never noticed this before, but there is a `systemd-rfkill` service. We don't need this since ConnMan will take care of this already. So we should be able to do the following:

    sudo systemctl mask systemd-rfkill.service
    sudo systemctl mask systemd-rfkill.socket

This should make sure that systemd doesn't try to mess with with the Bluetooth adapter.

**Comment by ylukin** [comment]
I was also seeing the same error (Operation not possible due to RF-Kill) on a brand new SD card with ev3dev-stretch-ev3-generic-2019-10-23.img. Masking systemd rfkill as suggested in previous comment worked for me. After several reboots, the brick connects to my Alexa device over bluetooth reliably now.

**Comment by laurensvalk** [comment]
Seeing similar issues, so reporting logs for reference in case this is helpful:

```
========== ev3dev-sysinfo ==========
Image file:         ev3dev-stretch-ev3-generic-2019-10-23 *
Kernel version:     4.14.117-ev3dev-2.3.4-ev3
Brickman:           0.10.2
BogoMIPS:           148.88
Bluetooth:          
Board:              board0
BOARD_INFO_HW_REV=7
BOARD_INFO_MODEL=LEGO MINDSTORMS EV3
BOARD_INFO_ROM_REV=6
BOARD_INFO_SERIAL_NUM=00165351FD02
BOARD_INFO_TYPE=main

```

> If we can get the Can't init device hci0: Operation not possible due to RF-kill error again, please run sudo journalctl --no-pager --boot=0 so we can see when Bluetooth was trying to start relative to other services.
```
sudo systemctl --no-pager status ev3-bluetooth -l
[sudo] password for robot: 
● ev3-bluetooth.service - LEGO MINDSTORMS EV3 built-in Bluetooth
   Loaded: loaded (/lib/systemd/system/ev3-bluetooth.service; static; vendor preset: enabled)
   Active: failed (Result: exit-code) since Sat 2019-12-14 17:53:55 UTC; 4 days ago
  Process: 470 ExecStart=/usr/bin/hciattach -t 30 /dev/ttyS2 texas 2000000 flow nosleep (code=exited, status=1/FAILURE)
  Process: 393 ExecStartPre=/bin/sleep 10 (code=exited, status=0/SUCCESS)

Dec 14 17:53:43 ev3dev systemd[1]: Starting LEGO MINDSTORMS EV3 built-in Bluetooth...
Dec 14 17:53:53 ev3dev hciattach[470]: Found a Texas Instruments' chip!
Dec 14 17:53:53 ev3dev hciattach[470]: Firmware file : /lib/firmware/ti-connectivity/TIInit_6.6.15.bts
Dec 14 17:53:53 ev3dev hciattach[470]: Loaded BTS script version 1
Dec 14 17:53:53 ev3dev hciattach[470]: texas: changing baud rate to 2000000, flow control to 1
Dec 14 17:53:55 ev3dev hciattach[470]: Can't init device hci0: Operation not possible due to RF-kill (132)Can't initialize device: Operation not possible due to RF-kill
Dec 14 17:53:55 ev3dev systemd[1]: ev3-bluetooth.service: Control process exited, code=exited status=1
Dec 14 17:53:55 ev3dev systemd[1]: Failed to start LEGO MINDSTORMS EV3 built-in Bluetooth.
Dec 14 17:53:55 ev3dev systemd[1]: ev3-bluetooth.service: Unit entered failed state.
Dec 14 17:53:55 ev3dev systemd[1]: ev3-bluetooth.service: Failed with result 'exit-code'.
```

> So we should be able to do the following:
```
sudo systemctl mask systemd-rfkill.service
sudo systemctl mask systemd-rfkill.socket
```

I will try this and report back.

Will be updating as I go. Leaderboard:

Successful:

- [x] .
- [x] .
- [x] .
- [x] .
- [x] .
- [x] .
- [x] .

Fail:

- [ ] .
- [ ] .
- [ ] .
- [ ] .
- [ ] .
- [ ] .
- [ ] .

**Comment by mbr4477** [comment]
@laurensvalk I can confirm that 
```
sudo systemctl mask systemd-rfkill.service
sudo systemctl mask systemd-rfkill.socket
```
plus a reboot worked for me. Thanks!

**Comment by laurensvalk** [comment]
Just coming back to say that Bluetooth hasn't failed a single time since :tada: Awesome work!

**Comment by CubixSystem** [comment]
@dlech Big thanks for fix!
I can confirm https://github.com/ev3dev/ev3dev/issues/1314#issuecomment-568186729 too

**Comment by dlech** [comment]
ev3dev-stretch R3 has been released with this fix. Please open a new issue if there are problems with the R3 release.
